internal/user/repository/eventsourcing/user.go

```diff
@@ -0,0 +1,345 @@
+package eventsourcing
+
+import (
+	"context"
+	"github.com/caos/zitadel/internal/errors"
+	es_models "github.com/caos/zitadel/internal/eventstore/models"
+	"github.com/caos/zitadel/internal/user/repository/eventsourcing/model"
+)
+
+func UserByIDQuery(id string, latestSequence uint64) (*es_models.SearchQuery, error) {
+	if id == "" {
+		return nil, errors.ThrowPreconditionFailed(nil, "EVENT-d8isw", "id should be filled")
+	}
+	return UserQuery(latestSequence).
+		AggregateIDFilter(id), nil
+}
+
+func UserQuery(latestSequence uint64) *es_models.SearchQuery {
+	return es_models.NewSearchQuery().
+		AggregateTypeFilter(model.UserAggregate).
+		LatestSequenceFilter(latestSequence)
+}
+
+func UserAggregate(ctx context.Context, aggCreator *es_models.AggregateCreator, user *model.User) (*es_models.Aggregate, error) {
+	if user == nil {
+		return nil, errors.ThrowPreconditionFailed(nil, "EVENT-dis83", "existing user should not be nil")
+	}
```

Comment:
I think we decided that this never can happen? Because we always check user == nil before we call the Aggregate-funcs

Reply:
also in the following funcs

Reply:
i think we said email on a user can never be nil, in my opinion this check is ok

Reply:
in my opinion no. because its validated in `eventstore.go` if the validation is wrong it's okay to panic